users/router.js

```diff
@@ -101,45 +57,74 @@ router.post('/', (req, res) => {
 // if we're creating users. keep in mind, you can also
 // verify this in the Mongo shell.
 router.get('/', (req, res) => {
-  return User
-    .find()
-    .exec()
+  return User.find().exec()
     .then(users => res.json(users.map(user => user.apiRepr())))
     .catch(err => console.log(err) && res.status(500).json({message: 'Internal server error'}));
 });
 
-
 // NB: at time of writing, passport uses callbacks, not promises
 const basicStrategy = new BasicStrategy(function(username, password, callback) {
-  let user;
-  User
-    .findOne({username: username})
-    .exec()
-    .then(_user => {
-      user = _user;
+  User.findOne({username: username}).exec()
+    .then(user => {
       if (!user) {
-        return callback(null, false, {message: 'Incorrect username'});
-      }
-      return user.validatePassword(password);
-    })
-    .then(isValid => {
-      if (!isValid) {
-        return callback(null, false, {message: 'Incorrect password'});
+        return callback(null, false);
       }
-      else {
-        return callback(null, user)
-      }
-    });
+
+      user.validatePassword(password)
```

Comment:
It caused an error when the username is not valid but cause `callback(null, false, {message: 'Incorrect username'});` returns an empty string to the .then(). Might be another way to fix it but I'll need to look into it